core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/parking/WaitParkComparison.java

```diff
@@ -0,0 +1,34 @@
+package com.baeldung.concurrent.parking;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.*;
+import java.util.concurrent.locks.LockSupport;
+
+class WaitParkComparison {
+    private static final Logger LOG = LoggerFactory.getLogger(WaitParkComparison.class);
+
+    public void parkingThread() throws InterruptedException {
+        Object lockObject = new Object();
+        Runnable task1 = () -> {
+            synchronized (lockObject) {
+                LOG.debug("Thread 1 is blocked");
+                LockSupport.park();
+                LOG.debug("Thread 1 resumed");
+            }
+        };
+        Thread thread1 = new Thread(task1);
+        thread1.start();
+        Runnable task2 = () -> {
+            LOG.debug("Thread 2 running ");
+            synchronized (lockObject) {
```

Comment:
As this is multi-threaded code, there's a chance that thread 2 will reach this point before thread 1 is parked,